raven/versioning.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import absolute_import
+
+__all__ = ('fetch_git_sha', 'fetch_package_version')
+
+import os.path
+import pkg_resources
+
+from .exceptions import InvalidGitRepository
+
+
+def fetch_git_sha(path, head='master'):
+    """
+    >>> fetch_git_sha(os.path.dirname(__file__))
+    """
+    revision_file = os.path.join(path, '.git', 'refs', 'heads', 'master')
```

Comment:
@dcramer 
I suggest using "HEAD" here instead (via `.git/HEAD`).

This might be (at least) a sha, or a link to a ref, e.g. `ref: refs/heads/develop`.

Even better would be to use `git describe --dirty --always`, but that requires the `git` binary, which might not be a given.